pkg/visor/init.go

```diff
@@ -212,9 +212,20 @@ func initDmsg(ctx context.Context, v *Visor, log *logging.Logger) error {
 	}
 	dmsgC := dmsgc.New(v.conf.PK, v.conf.SK, v.ebc, v.conf.Dmsg)
 
-	time.Sleep(200 * time.Millisecond)
-	go dmsgC.Serve(context.Background())
-	time.Sleep(200 * time.Millisecond)
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		dmsgC.Serve(context.Background())
+	}()
+
+	v.pushCloseStack("dmsg", func() error {
+		if err := dmsgC.Close(); err != nil {
```

Comment:
We are closing twice. Once here and once when we shutown the clients on https://github.com/skycoin/skywire/blob/2a76c6a19326ff8f78d791d740252895ac15be6a/pkg/transport/manager.go?plain=1#L547 if I am not mistaken.

Reply:
Will have to test this code properly. Currently the `Close()` from `dmsgC` is not being called.